qa/integration-tests/src/test/java/io/camunda/it/client/DecisionInstanceQueryTest.java

```diff
@@ -0,0 +1,274 @@
+  }
+
+  @TestTemplate
+  public void shouldRetrieveDecisionInstanceByDecisionKeyFilterIn(final ZeebeClient zeebeClient) {
+    // when
+    final long decisionDefinitionKey =
+        evaluatedDecisions.get(DECISION_DEFINITION_ID_1).getDecisionKey();
+    final BasicLongFilterProperty filter = new BasicLongFilterProperty();
+    filter.set$In(List.of(Long.MAX_VALUE, decisionDefinitionKey));
+    final var result =
+        zeebeClient
+            .newDecisionInstanceQuery()
+            .filter(f -> f.decisionDefinitionKey(filter))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(1);
+    assertThat(result.items().getFirst().getDecisionDefinitionKey())
+        .isEqualTo(decisionDefinitionKey);
+    assertThat(result.items().getFirst().getDecisionInstanceKey())
+        .isEqualTo(evaluatedDecisions.get(DECISION_DEFINITION_ID_1).getDecisionInstanceKey());
+  }
+
+  @TestTemplate
+  public void shouldRetrieveDecisionInstanceByDecisionInstanceKey(final ZeebeClient zeebeClient) {
+    // when
+    final long decisionInstanceKey =
+        evaluatedDecisions.get(DECISION_DEFINITION_ID_2).getDecisionInstanceKey();
+    final var result =
+        zeebeClient
+            .newDecisionInstanceQuery()
+            .filter(f -> f.decisionInstanceKey(decisionInstanceKey))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(2);
+    assertThat(result.items().getFirst().getDecisionInstanceKey()).isEqualTo(decisionInstanceKey);
+    assertThat(result.items().getLast().getDecisionInstanceKey()).isEqualTo(decisionInstanceKey);
+  }
+
+  @TestTemplate
+  public void shouldRetrieveDecisionInstanceByStateAndType(final ZeebeClient zeebeClient) {
+    // when
+    final DecisionInstanceState state = DecisionInstanceState.EVALUATED;
+    final DecisionDefinitionType type = DecisionDefinitionType.DECISION_TABLE;
+    final var result =
+        zeebeClient
+            .newDecisionInstanceQuery()
+            .filter(f -> f.state(state).decisionDefinitionType(type))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(3);
+  }
+
+  @TestTemplate
+  public void shouldRetrieveDecisionInstanceByEvaluationDate(final ZeebeClient zeebeClient) {
+    // given
+    final var allResult =
+        zeebeClient.newDecisionInstanceQuery().page(p -> p.limit(1)).send().join();
+    final var di = allResult.items().getFirst();
+
+    // when
+    final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXX");
+    final var offsetDateTime = OffsetDateTime.parse(di.getEvaluationDate(), formatter);
+    final var result =
+        zeebeClient
+            .newDecisionInstanceQuery()
+            .filter(f -> f.evaluationDate(offsetDateTime.toString()))
+            .send()
+            .join();
```

Comment:
💭 By the way, @danielkelemen is currently building [this here](https://github.com/camunda/camunda/pull/24760), which should make it a lot nicer to write this 😃